rundeckapp/src/test/groovy/rundeck/controllers/ScheduledExecutionControllerSpec.groovy

```diff
@@ -3909,4 +3913,113 @@ class ScheduledExecutionControllerSpec extends RundeckHibernateSpec implements C
         'https://web.server/geto'                                   |_
         'http://web.server/geto'                                    |_
     }
+
+    def "test remote url basic auth"() {
+        given:
+        controller.apiService = Mock(ApiService)
+        controller.frameworkService = Mock(FrameworkService)
+
+        def url = 'http://web.server/geto'
+
+        def jobChangeLogger = Mock(Logger)
+        controller.logger = jobChangeLogger
+
+        JobOptionConfigRemoteUrl configRemoteUrl = new JobOptionConfigRemoteUrl()
+        configRemoteUrl.authenticationType = RemoteUrlAuthenticationType.BASIC
+        configRemoteUrl.username = "test"
+        configRemoteUrl.password ="test"
+
+        when:
+        controller.getRemoteJSON(url, configRemoteUrl, 100, 100, 5,false)
+
+        then:
+        1 * controller.logger.debug("getRemoteJSON using authentication")
+        1 * controller.logger.debug("Basic authentication with user test")
+
+        def e = thrown(UnknownHostException)
+        e.message.contains("web.server")
+
+    }
+
+    def "test remote url api-token auth"() {
+        given:
+        controller.apiService = Mock(ApiService)
+        controller.frameworkService = Mock(FrameworkService)
+
+        def url = 'http://web.server/geto'
+
+        def jobChangeLogger = Mock(Logger)
+        controller.logger = jobChangeLogger
+
+        JobOptionConfigRemoteUrl configRemoteUrl = new JobOptionConfigRemoteUrl()
+        configRemoteUrl.authenticationType = RemoteUrlAuthenticationType.API_KEY
+        configRemoteUrl.token = "123"
+        configRemoteUrl.keyName = "apiKey"
+        configRemoteUrl.apiTokenReporter = ApiTokenReporter.HEADER
+
+        when:
+        controller.getRemoteJSON(url, configRemoteUrl, 100, 100, 5,false)
+
+        then:
+        1 * controller.logger.debug("getRemoteJSON using authentication")
+        1 * controller.logger.debug("API key authentication with Header apiKey")
```

Comment:
Should avoid relying on side-effect of using the logger.debug in the test.

Perhaps introduce an interface that can be mocked.  E.g. `interface HttpClientCreator{ HttpClient<HttpResponse> createClient() }` then you can replace [this line](https://github.com/rundeck/rundeck/pull/8193/files#diff-0588b6c107c25b5c990676b0df69cfec52009719523a901e54dd5eb9feb91622R850) with a call to the HttpClientCreator. 

Then you can test the `setUri`, `setHeader` etc all have the right values